### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by sullerandras**

### CODE REVIEW

**Comment by ST-DDT** [comment]
I will look at it in detail tomorrow.
Looks good to me, i will probably suggest some variable renames to make it more consistent throughout the mod.

**Comment by sullerandras** [comment]
@ST-DDT To make collaboration a bit easier, i invited you as a collaborator to this github repo so you should be able to push commits directly into this PR.

**Review by ST-DDT — APPROVED** [review]
I did all renames I had in mind and also fixed the open TODOs.

**Comment by ST-DDT** [comment]
Ah during playtesting just now I encountered some labels that did not clean up after the owner was destroyed and they lived until their next update (which based on factory size may take a while)

Maybe we need to subscribe to a destroy event as well?

lonelylabel

File: control.lua [hunk 1]
```diff
@@ -1,179 +1,496 @@
...
+  if global.entity_data ~= nil then
+    for _, data in pairs(global.entity_data) do
+      remove_label(data)
+    end
+  end
+  global.entity_data = {}
+  global.iteration = 0
+  global.show_labels = settings.global["utilization-monitor-show-labels"].value
+  global.entities_per_tick = settings.global["utilization-monitor-entities-per-tick"].value
+  global.iterations_per_update = settings.global["utilization-monitor-entities-per-tick"].value
```
Comment (sullerandras) [inline control.lua]: Isn't this supposed to be `utilization-monitor-iterations-per-update` here?

**Comment by sullerandras** [comment]
Yes i think we need to subscribe for destroy events as well, so we know when to destroy the labels.

File: control.lua (hunk 1 above)
Comment (ST-DDT) [reply control.lua]: Oh yes, thanks. Fixed!

**Comment by ST-DDT** [comment]
> Yes i think we need to subscribe for destroy events as well, so we know when to destroy the labels.

I added 3 of them. I hope that covers them all.

````lua
script.on_event({defines.events.on_entity_died, defines.events.on_player_mined_entity, defines.events.on_robot_mined_entity}, on_destroyed)
````

Anything else missing?

**Comment by sullerandras** [comment]
I'll release it soon.

**Pull request merged**